tests/unit/test_utils.py

```diff
@@ -0,0 +1,336 @@
+from pathlib import Path, PosixPath
+from shutil import rmtree
+from typing import Generator, Tuple, List
+from zipfile import ZipFile
+
+import os
+import pytest
+import requests_mock
+import typer
+from snowcli import utils
+
+from tests.unit.test_data.test_data import *
+
+# TODO: check for consistency in using ' or "
+
+
+class TestUtils:
+    APP_ZIP = "app.zip"
+    CORRECT_METADATA = "correct_metadata.yaml"
+    FILE_IN_A_SUBDIR = "file_in_a_subdir.txt"
+    FILE_IN_SECOND_TEST_DIRECTORY = "very_important_file.txt"
+    REQUIREMENTS_SNOWFLAKE = "requirements.snowflake.txt"
+    REQUIREMENTS_TXT = "requirements.txt"
+    SECOND_TEST_DIRECTORY = "other_test_dir"
+    SUBDIR = "subdir"
+    TEMP_DIR_FOR_APP_ZIP = "temp_dir"
+    TEMP_TEST_DIRECTORY = ".tests"
+
+    @pytest.mark.parametrize("argument", utils.YesNoAskOptions)
+    def test_yes_no_ask_callback_with_correct_argument(self, argument: str):
+        result = utils.yes_no_ask_callback(argument)
+
+        assert result == argument
+
+    @pytest.mark.parametrize("argument", bad_arguments_for_yesnoask)
+    def test_yes_no_ask_callback_with_incorrect_argument(self, argument):
+        with pytest.raises(typer.BadParameter) as e_info:
+            utils.yes_no_ask_callback(argument)
+
+        assert (
+            e_info.value.message
+            == f"Valid values: ['yes', 'no', 'ask']. You provided: {argument}"
+        )
+
+    @pytest.mark.parametrize("arguments", positive_arguments_for_deploy_names)
+    def test_get_deploy_names_correct(
+        self, arguments: Tuple[Tuple[str, str, str], dict]
+    ):
+        result = utils.get_deploy_names(*arguments[0])
+
+        assert result == arguments[1]
+
+    def test_prepare_app_zip(
+        self,
+        temp_test_directory: str,
+        correct_app_zip: str,
+        temp_directory_for_app_zip: str,
+    ):
+        result = utils.prepare_app_zip(correct_app_zip, temp_directory_for_app_zip)
+
+        assert result == temp_directory_for_app_zip + "/app.zip"
+
+    def test_prepare_app_zip_if_exception_is_raised_if_no_source(
+        self, temp_directory_for_app_zip
```

Comment:
Why not `tmp_path`? https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html

Reply:
tmp_path is designed for use inside single test. Every fixture we currently use is called 31(max) to 2(min) times - so there is no point. 
We could use tmp_path_factory, which is more suitable for fixtures, but introducing it would change exactly one line in every fixture code, and i can`t see any gain from it.